### ISSUE TITLE
Elasticseach out of memory

### ISSUE THREAD

**Issue by freignat91** [appcelerator-archive/amp#1382]
On cluster Ireland, after 7 days ES stops to work for good, it tried to restart for time to time each time failed with logs:

```
amp_elasticsearch.1.mkpcvhxo494l@fr-amp-manager1    | java.lang.OutOfMemoryError: GC overhead limit exceeded

amp_elasticsearch.1.mkpcvhxo494l@fr-amp-manager1    | [2017-06-13T15:55:46,952][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [] fatal error in thread [elasticsearch[wZnfCVR][refresh][T#1]], exiting

amp_elasticsearch.1.mkpcvhxo494l@fr-amp-manager1    | java.lang.OutOfMemoryError: GC overhead limit exceeded
```

It was on nodes with 8GB of memory
We need to add tools to delete/store old metrics/logs with Curator, Elastalert or something equivalent

**Comment by ndegory** [comment]
depends on what's on the same node, was ES alone on this node? other containers?
depending on your swarm node labeling strategy, it makes a difference.
There's a default setting for the java heap size that should work fine most of the time, but it can be tweaked to your specific needs: set JAVA_HEAP_SIZE (value in MB) as a service environment variable. Please check the README.md for mode details.

**Comment by ndegory** [comment]
but regarding tools to manage the retention policy, I agree, it's mandatory we add them asap. Are you on it?

**Comment by freignat91** [comment]
no, not before the 1.0

**Comment by ndegory** [comment]
1.0 is not planned yet, will it be in 0.13?

**Comment by ndegory** [comment]
not seeing this issue anymore on long running clusters, so I'll close this issue as it's been addressed in 0.12.0. The monitoring aspect of it should be the object of another issue/PR.
